packages/material-ui-utils/src/deepmerge.ts

```diff
@@ -26,7 +26,7 @@ export default function deepmerge<T>(
         return;
       }
 
-      if (isPlainObject(source[key]) && key in target) {
+      if (isPlainObject(source[key]) && key in target && isPlainObject(target[key])) {
```

Comment:
We need `deepmerge` only if both source and target are objects. Otherwise we can override with `source[key]`